src/main/java/org/opensearch/geospatial/ip2geo/common/InputFormatValidator.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.geospatial.ip2geo.common;
+
+import java.io.UnsupportedEncodingException;
+
+import org.opensearch.OpenSearchException;
+import org.opensearch.common.Strings;
+
+public class InputFormatValidator {
+    private static final int MAX_DATASOURCE_NAME_BYTES = 127;
+
+    public String validateDatasourceName(final String datasourceName) {
+        if (datasourceName == null || datasourceName.isEmpty()) {
+            return "datasource name must not be empty";
+        }
+        if (!Strings.validFileName(datasourceName)) {
+            return "datasource name must not contain the following characters " + Strings.INVALID_FILENAME_CHARS;
+        }
+        if (datasourceName.contains("#")) {
+            return "datasource name must not contain '#'";
+        }
+        if (datasourceName.contains(":")) {
+            return "datasource name must not contain ':'";
+        }
+        if (datasourceName.charAt(0) == '_' || datasourceName.charAt(0) == '-' || datasourceName.charAt(0) == '+') {
+            return "datasource name must not start with '_', '-', or '+'";
+        }
+        int byteCount = 0;
+        try {
+            byteCount = datasourceName.getBytes("UTF-8").length;
```

Comment:
we can use StandardCharsets.UTF_8 instead of using own string constant

Reply:
ack